### PR TITLE
Split Cart component into functional (Cart) and view parts (CartView)

### PR DESCRIPTION
**Pull request by yurii-ve**
- created `CartView` component
- updated `Cart` component to handle all the logic and render `CartView`
- removed `CartItemView` component as `CartItem` became purely view (logic moved to `Cart`)

Notes:
 - this PR is the first part of the change. next PR will contain changes to the `Cart` board
 - all JSX structure and styles were just moved around, changes to them are not responsibility of this PR

### CODE REVIEW

File: src/components/cart/cart-item/cart-item.tsx [hunk 1]
```diff
@@ -1,39 +1,75 @@
...
     isLast?: boolean;
     cartItem: cart.LineItem;
+    onRemoveButtonClick: () => void;
+    onQuantityChange: (newQuantity: number) => void;
 }
 
-export const CartItem = ({ cartItem, className, isLast }: CartItemProps) => {
-    const { trigger: updateQuantity } = useUpdateCartItemQuantity();
-    const { trigger: removeItem } = useRemoveItemFromCart();
+export const CartItem = ({ cartItem, className, isLast, onRemoveButtonClick, onQuantityChange }: CartItemProps) => {
```
Comment (VladyslavG) [inline src/components/cart/cart-item/cart-item.tsx]: by the way function `onRemoveButtonClick` and `onQuantityChange` have inconsistent names
it should be then `onRemove` + `onQuantityChange`

File: src/components/cart/cart.tsx [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
 export const Cart = () => {
+    const ecomAPI = useEcomAPI();
     const { isOpen, setIsOpen } = useCartOpen();
     const { data: cart } = useCart();
     const { data: cartTotals } = useCartTotals();
-    const [checkoutAttempted, setCheckoutAttempted] = useState(false);
-    const ecomAPI = useEcomAPI();
+    const { trigger: updateQuantity } = useUpdateCartItemQuantity();
+    const { trigger: removeItem } = useRemoveItemFromCart();
+    const [error, setError] = useState<string>();
```
Comment (VladyslavG) [inline src/components/cart/cart.tsx]: is `error` a new functionality in this PR ?

File: src/components/cart/cart.tsx [hunk 3]
```diff
@@ -32,33 +31,18 @@ export const Cart = () => {
...
-                    </div>
-                </div>
-            )}
+            <CartView
+                cart={cart!}
+                cartTotals={cartTotals}
+                errorMessage={error}
+                onCheckout={handleCheckout}
+                onItemRemove={(id) => removeItem(id)}
+                onItemQuantityChange={(id, quantity) => updateQuantity({ id, quantity })}
```
Comment (VladyslavG) [inline src/components/cart/cart.tsx]: let's change type of prop onItemQuantityChange to accept updateQuantity without any param changes so we can write

`onItemQuantityChange = {updateQuantity }`

File: src/components/cart/cart.tsx [hunk 4]
```diff
@@ -32,33 +31,18 @@ export const Cart = () => {
...
-                        </button>
-                    </div>
-                </div>
-            )}
+            <CartView
+                cart={cart!}
+                cartTotals={cartTotals}
+                errorMessage={error}
+                onCheckout={handleCheckout}
+                onItemRemove={(id) => removeItem(id)}
```
Comment (VladyslavG) [inline src/components/cart/cart.tsx]: onItemRemove={removeItem}

File: src/components/cart/cart-view/cart-view.tsx [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    const isEmpty = !cart?.lineItems || cart.lineItems.length === 0;
+
+    if (isEmpty) {
+        return <div className={styles.emptyCart}>Cart is empty</div>;
+    }
+
+    return (
+        <div className={styles.cart}>
+            <div className={styles.items}>
+                {cart?.lineItems?.map((item) => (
```
Comment (VladyslavG) [inline src/components/cart/cart-view/cart-view.tsx]: why do you add `?` after cart everywhere ? it is typed as `cart: Cart;`

File: src/components/cart/cart.tsx (hunk 2 above)
Comment (yurii-ve) [reply src/components/cart/cart.tsx]: it was an experiment.. good catch

**Review by olehrakwix — APPROVED** [review]

File: src/components/cart/cart.tsx (hunk 2 above)
Comment (VladyslavG) [reply src/components/cart/cart.tsx]: or it was a test :)

**Pull request merged**